filament/src/details/Scene.cpp

```diff
@@ -347,6 +351,12 @@ void FScene::prepareVisibleRenderables(Range<uint32_t> visibleRenderables) noexc
         uboData.userData = sceneData.elementAt<USER_DATA>(i);
 
         mHasContactShadows = mHasContactShadows || visibility.screenSpaceContactShadows;
+
+        // Prepare the InstanceBuffer for this Renderable.
+        // TODO: maybe do this inside updateUBOs? And avoid updating the actual UBO?
+        if (UTILS_UNLIKELY(instanceBuffer)) {
+            instanceBuffer->prepare(mEngine, model, uboData);
+        }
```

Comment:
this should be done after culling otherwise we're doing thing for everything in the scene, even if it's not visible.

Reply:
Doesn't `prepareVisibleRenderables` happen after culling already?

Reply:
oh yes it does, sorry I misread the code